modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-ack.adoc

```diff
@@ -0,0 +1,153 @@
+
+==== Acknowledgment Timeout with MANUAL Acknowledge
+
+When consuming a message using `MANUAL` acknowledgment mode in the Subscriber or Consume operations, the app must execute either an ACK or NACK operation using the message `ackToken` before the `acknowledgementTimeout` limit is reached. If neither operation executes with the given token before the timeout, the message automatically returns to the queue for redelivery. Any attempt to acknowledge the message after the timeout limit fails due to an expired token.
+
+== Manual Acknowledgment Using the ACK Operation
+
+Acknowledging a message informs the broker that the message has been processed and must be removed from the queue to prevent redelivery.
+
+The acknowledgment token is a unique identifier for the message that you must use when performing the ACK operation. It is available in the `ackToken` field of the message attributes. For example:
+
+[source,xml,linenums]
+----
+<flow name="consumerWithManualAck">
+    <scheduler>
+        <scheduling-strategy >
+          <fixed-frequency />
+        </scheduling-strategy>
+    </scheduler>
+
+    <!-- Consume a message -->
+    <anypoint-mq:consume destination="${destination}"
+         acknowledgementMode="MANUAL"
+            config-ref="Anypoint_MQ_Config"/>
+
+    <!-- Save the ackToken for future acknowledgment -->
+    <set-variable variableName="currentAckToken"
+       value="#[attributes.ackToken]"/>
+
+    <!-- Do the required processing of the message -->
+    <flow-ref name="process-message"/>
+
+    <!-- Use the ackToken to acknowledge the message -->
+    <anypoint-mq:ack ackToken="#[vars.currentAckToken]"
+         config-ref="Anypoint_MQ_Config" />
+</flow>
+----
+
+If a non-void operation is invoked during message processing, the payload and attributes of the Mule message are modified. To perform an ACK operation after processing, you must save the `ackToken` in a variable.
+
+To save attributes to use later, use the `target` and `targetValue` parameters to store the whole message in a variable:
+
+[source,xml,linenums]
+----
+<flow name="consumerWithManualAck">
+      <scheduler>
+        <scheduling-strategy >
+          <fixed-frequency />
+        </scheduling-strategy>
+      </scheduler>
+
+    <!-- Consume a message -->
+    <anypoint-mq:consume destination="${destination}"
+        acknowledgementMode="MANUAL"
+        config-ref="Anypoint_MQ_Config"
+        target="mqMessage"
+        targetValue="#[message]"/>
+
+
+    <!--Do any message processing-->
+    <jms:publish-consume destination="#[vars.mqMessage.attributes.targetDestination]"
+        config-ref="JMS_Config">
+        <jms:message>
+            <jms:body>#[vars.mqMessage.payload]</jms:body>
+        </jms:message>
+    </jms:publish-consume>
+
+    <!-- Use the ackToken to ACK the message -->
+    <anypoint-mq:ack ackToken="#[vars.mqMessage.attributes.ackToken]"
+        config-ref="Anypoint_MQ_Config" />
+</flow>
+----
+
+=== Execute a NACK
```

Comment:
Is this really a subtopic to Manual Acknowledgment Using the ACK Operation?

Also, the use of an imperative phrase ("Execute...") leads me to expect a task rather than an example, which is more of a reference content.